gcs/blockcf2/maxsize_test.go

```diff
@@ -0,0 +1,268 @@
+package blockcf2
+
+import (
+	"encoding/binary"
+	"math/bits"
+	"math/rand"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/dchest/siphash"
+	"github.com/decred/dcrd/chaincfg/v3"
+	"github.com/decred/dcrd/gcs/v4"
+	"github.com/decred/dcrd/wire"
+)
+
+// TestMaxSize verifies the max size of blockcf2 filters for various
+// parameters.
+//
+// This test is meant to nail down the values generated by various
+// package-level constants to known good hardcoded values such that if any of
+// the values change, this test breaks and any code that relies on assumptions
+// about the max filter size having a particular value are reviewed.
+func TestMaxSize(t *testing.T) {
+
+	// entrySizeForSize returns the size of scripts that ensures that every
+	// entry on a filter may be unique (i.e. no two scripts will generate
+	// the same internal value after siphash and modulo reductions).
+	entrySizeForSize := func(size uint32) uint32 {
```

Comment:
In practice, for all relevant tested sizes, the script length that maximizes the cfilters will be 3.